File: src/test/java/com/example/dpop/DpopApplicationTests.java
package com.example.dpop;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.modulith.core.ApplicationModules;
import org.springframework.test.context.ActiveProfiles;

@SpringBootTest
@ActiveProfiles("test")
class DpopApplicationTests {

    @Test
    void contextLoads() {
    }

    @Test
    void modulithStructureIsValid() {
        var modules = ApplicationModules.of(DpopApplication.class);
        modules.verify();
    }
}

File: src/test/kotlin/com/example/dpop/DpopApplicationTests.kt
package com.example.dpop

import io.kotest.core.spec.style.BehaviorSpec
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.modulith.core.ApplicationModules
import org.springframework.test.context.ActiveProfiles

@SpringBootTest
@ActiveProfiles("test")
class DpopApplicationTests : BehaviorSpec({

    given("the Spring application context") {
        then("it loads successfully") {
        }

        then("the modulith structure is valid") {
            val modules = ApplicationModules.of(DpopApplication::class.java)
            modules.verify()
        }
    }
})
